### PR TITLE
[FrameworkBundle][HttpClient] Adding an explanation of `ThrottlingHttpClient` integration with the framework

### PR DESCRIPTION
**Pull request by maxhelias**

### CODE REVIEW

**Comment by maxhelias** [comment]
If anyone can help me with the CI, I'd be happy.

File: http_client.rst [hunk 1]
```diff
@@ -1488,30 +1488,108 @@ Limit the Number of Requests
 ----------------------------
 
 This component provides a :class:`Symfony\\Component\\HttpClient\\ThrottlingHttpClient`
-decorator that allows to limit the number of requests within a certain period.
+decorator that allows to limit the number of requests within a certain period, potentially delaying calls based on the rate limiting policy.
```
Comment (nicolas-grekas) [inline http_client.rst]:
```suggestion
decorator that allows to limit the number of requests within a certain period,
potentially delaying calls based on the rate limiting policy.
```

File: http_client.rst [hunk 2]
```diff
@@ -1488,30 +1488,108 @@ Limit the Number of Requests
...
-    $requests = [];
-    for ($i = 0; $i < 100; $i++) {
-        $requests[] = $client->request('GET', 'https://example.com');
-    }
+        # config/packages/framework.yaml
+        framework:
+            http_client:
+                scoped_clients:
+                    symfony.client:
+                        base_uri: 'https://symfony.com'
```
Comment (nicolas-grekas) [inline http_client.rst]:
```suggestion
                        base_uri: 'https://example.com'
```

File: http_client.rst [hunk 3]
```diff
@@ -1488,30 +1488,108 @@ Limit the Number of Requests
...
+                        rate_limiter: 'http_symfony_termsofuse'
```
Comment (nicolas-grekas) [inline http_client.rst]:
```suggestion
                        rate_limiter: 'http_example_limiter'
```

File: http_client.rst [hunk 4]
```diff
@@ -1488,30 +1488,108 @@ Limit the Number of Requests
...
+                ;
+        };
+
+    .. code-block:: php-standalone
+
+        use Symfony\Component\HttpClient\HttpClient;
+        use Symfony\Component\HttpClient\ThrottlingHttpClient;
+        use Symfony\Component\RateLimiter\LimiterInterface;
+
+        $rateLimiter = ...; // $rateLimiter is an instance of Symfony\Component\RateLimiter\LimiterInterface
```
Comment (nicolas-grekas) [inline http_client.rst]: how involving would it be to show a working code here?

File: http_client.rst [hunk 5]
```diff
@@ -1488,30 +1488,108 @@ Limit the Number of Requests
...
+        $client = HttpClient::create();
+        $client = new ScopingHttpClient($client, [
+            'base_uri' => 'https://symfony.com',
+        ]);
```
Comment (nicolas-grekas) [inline http_client.rst]: do we need the scoping client here?
```suggestion
        $client = HttpClient::create();
```

Comment (nicolas-grekas) [reply http_client.rst]: note that creating a ScopingHttpClient with a base URI should be done with the forBaseUri static method.

File: http_client.rst (hunk 4 above)
Comment (maxhelias) [reply http_client.rst]: I just got lazy and used the existing PHP example 😅 I'll update with a concrete example.

File: http_client.rst (hunk 5 above)
Comment (maxhelias) [reply http_client.rst]: Not really, it's more to ensure that the examples between framework and standalone integration are strictly identical

File: http_client.rst [hunk 6]
```diff
@@ -1488,30 +1488,104 @@ Limit the Number of Requests
...
+        $factory = new RateLimiterFactory([
+            'id' => 'http_example_limiter',
+            'policy' => 'token_bucket',
+            'limit' => 10,
+            'rate' => ['interval' => '5 seconds', 'amount' => 10],
+        ], new InMemoryStorage());
+        $limiter = $factory->create();
+
+        $client = HttpClient::createForBaseUri('https://example.com');
+        $client = new ThrottlingHttpClient($client, $limiter);
```
Comment (OskarStark) [inline http_client.rst]: Hmm I don't like the two `$client` variables 🤔

**Review by nicolas-grekas — APPROVED** [review]

File: http_client.rst [hunk 7]
```diff
@@ -1488,30 +1488,104 @@ Limit the Number of Requests
...
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:http-client>
+                    <framework:scoped-client name="symfony.client"
+                        base-uri="https://symfony.com"
```
Comment (xabbuh) [inline http_client.rst]:
```suggestion
                        base-uri="https://example.com"
```

File: http_client.rst [hunk 8]
```diff
@@ -1488,30 +1488,104 @@ Limit the Number of Requests
...
+                    />
+                </framework:http-client>
+
+                <framework:rate-limiter>
+                    <!-- Don't send more than 10 requests in 5 seconds -->
+                    <framework:limiter name="http_example_limiter"
+                        policy="token_bucket"
+                        limit="10"
+                    >
+                        <framework:rate interval="5 seconds" amout="10"/>
```
Comment (xabbuh) [inline http_client.rst]:
```suggestion
                        <framework:rate interval="5 seconds" amount="10"/>
```

**Comment by javiereguiluz** [comment]
Maxime, thanks for this contribution. These are good changes. Thanks to reviewers too!

**Pull request merged**